### PR TITLE
[feat] Capture stdout/stderr logs

### PR DESCRIPTION
**Pull request by mihran113**

### CODE REVIEW

**Comment by mihran113** [comment]
Resolves #1659

File: aim/ext/resource/tracker.py [hunk 1]
```diff
@@ -23,16 +73,26 @@ def reset_proc_interval():
         """
         cpu_percent(0.0)
 
-    def __init__(self, track, interval: Union[int, float] = STAT_INTERVAL_DEFAULT):
+    def __init__(self, track, interval: Union[int, float] = STAT_INTERVAL_DEFAULT, capture_logs: bool = True):
         self._track_func = weakref.WeakMethod(track)
-        self.interval = interval
+        self._interval = None
```
[the remark this answers is not in the document]
Comment (mihran113) [reply aim/ext/resource/tracker.py:82]: will change

File: aim/ext/resource/tracker.py [hunk 2]
```diff
@@ -73,6 +123,14 @@ def stop(self):
...
         self._shutdown = True
         self._th_collector.join()
+        if self._capture_logs:
+            # read and store remaining buffered logs
+            self._store_buffered_logs()
+            # unregister the buffer
+            del self._buffer_registry[id(self)]
+            # uninstall stream patching if no buffer is left in the registry
+            if not self._buffer_registry:
+                self._uninstall_stream_patches()
```
[the remark this answers is not in the document]
Comment (mihran113) [reply aim/ext/resource/tracker.py:137]: nothing in particular, as the buffers are kept in registry, and creation/destruction order isn't playing any role

**Review by alberttorosyan — APPROVED** [review]
lgtm

**Pull request merged**